src/coreclr/jit/morph.cpp

```diff
@@ -11839,21 +11839,21 @@ void Compiler::fgAssertionGen(GenTree* tree)
     //
     auto addImpliedAssertions = [=](AssertionIndex index, ASSERT_TP& assertions) {
         AssertionDsc* const assertion = optGetAssertion(index);
-        if ((assertion->assertionKind == OAK_EQUAL) && (assertion->op1.kind == O1K_LCLVAR) &&
+        if (optLocalAssertionProp && (assertion->assertionKind == OAK_EQUAL) && (assertion->op1.kind == O1K_LCLVAR) &&
```

Comment:
How are we calling this outside of global morph?

Reply:
@AndyAyersMS we don't, I forgot to remove this redundant check when I was investigating a bug 🙂 I'll push a commit to remove it in https://github.com/dotnet/runtime/pull/113920 clean up PR to avoid spinning CI again

Reply:
Pushed: https://github.com/dotnet/runtime/pull/113920/commits/97449fbb6e32cff12b6cd117a6cb21c047dcd245